### ISSUE TITLE
PostGIS Tiger: Data not loaded to use geocode function

### ISSUE THREAD

**Issue by jaredramirez** [supabase/supabase#1318]
# Feature request

## Is your feature request related to a problem? Please describe.

I have enabled to `POSTGIS_TIGER_GEOCODER`, but whenever I use the its geocoding functionality no results are returned. I believe this is because while the extension is enabled, the census data it uses to geocode is not loaded. See the PostGIS Tiger docs for more info: http://postgis.net/docs/postgis_installation.html#install_tiger_geocoder_extension.

The query I'm using to test is:
```
select tiger.geocode('... my address ...');
```
which returns
```
 geocode
---------
(0 rows)
```

## Describe the solution you'd like

Documentation on how to load the census data and/or automatically loading it when enabling the extension.

**Comment by steve-chavez** [comment]
Hey @jaredramirez,

Have you tried enabling the `ADDRESS_STANDARDIZER_DATA_US` extension?

**Comment by burggraf** [comment]
I've seen this issue myself when I tried to load that extension initially.  I'll look at this when we start to add documentation for the individual extensions.  We may need to use the new data loader utility to load data when this extension is loaded, but it might also be more complicated than that if the amount of data is larger than the free tier database max size.

**Comment by vbylen** [comment]
yeah, just wanted to report I'm encountering the same issue here. 

as an alternative I highly recommend checking out [geocode maps](https://geocode.maps.co)

**Comment by encima** [comment]
Closing this due to age and version changes that have occurred since opening.

Please do comment if you want it reopened and addressed!

**Comment by iveshenry18** [comment]
Hey! As far as I can tell, this is still an open issue. There seems to be a lot of confusion around this set of extensions, so  here's my current understanding of what's going on:

Supabase offers the `postgis` extension and a few related extensions: `address_standardizer`, `address_standardizer_data_us`, and `postgis_tiger_geocoder`. 
- `postgis` primarily provides the data structures and operators for working with geographical points (i.e. lat/long).
- `address_standardizer` provides functions for converting string addresses to parsed addresses.
- `address_standardizer_data_us` provides the data necessary for the `address_standardizer` functions to work with US addresses (specifically `gaz`, `lex`, and `rules` table, details [here](https://postgis.net/docs/manual-2.5/Address_Standardizer.html))
- `postgis_tiger_geocoder` _should_ provide geocoding (converting address to lat/long) and reverse-geocoding (converting lat/long to address) functionality. It uses US census data to do this.

The problem is that the data that powers `postgis_tiger_geocoder`, and even the tables that hold the data (within the `tiger_data` schema), **have to be retrieved via an install script**. Specifically, the extension provides a function that creates data loading shell scripts (given an OS and configuration), and those scripts must be run on the host server. These steps are documented in detail [here](https://postgis.net/docs/postgis_installation.html#install_tiger_geocoder_extension).

There is no apparent way to run these scripts within hosted Supabase, since we obviously don't have access to the host server shell. If that's correct, then the `postgis_tiger_geocoder` extension is basically an empty extension in Supabase.

I could be wrong about all this, so if I've missed an update or some documentation somewhere, please let me know! Otherwise, this is still an open issue. I'm not sure how feasible it is to support this type of workflow within Supabase's existing extension loading paradigm, but it's certainly functionality that would be valuable for many apps.

**Comment by vladmiller** [comment]
Oi lads, any update on this? 

As of now, I'm trying to set-up a workflow where I'll create all of the data in the local postgres container and then migrate `tiger_data` schema to Supabase DB. Not sure if this will work, but worth trying.

However, doing it on supabase would be way better.

**Comment by iveshenry18** [comment]
fwiw I gave up on trying to get this to work and moved to the Google Places API. Since our app just needed a fairly standard city autocomplete as a part of a small feature, it makes sense to just use a 3rd party for that and keep the core app in Supabase. If managing your own geographic data is central to your app, then yeah, I'd say Supabase is not ready to support your use case.

**Comment by Hallidayo** [comment]
Hi @steve-chavez & @burggraf - Just checking in on some older issues, Just tried to replicate this locally and I also get issues with geocoding using `postgis_tiger_geocoder`. Can we geocode within Supabase using the `postgis_tiger_geocoder` extension?

**Comment by wernermorgenstern** [comment]
I am having issue too.

When the `nation_script_load` script is run, I get this error:

```
CREATE TABLE tiger_data.state_all(CONSTRAINT pk_state_all PRIMARY KEY (statefp),CONSTRAINT uidx_state_all_stusps  UNIQUE (stusps), CONSTRAINT uidx_state_all_gid UNIQUE (gid) ) INHERITS(tiger.state);
ERROR:  must be owner of table state
```

**Comment by BaileySimrell** [comment]
Thanks for the detailed writeup here @iveshenry18 ... I think this captures just about everything I have ran into as well ... going to try my best to do a push of the data from local machine to Github to try and get postgis_tiger_geocoder working one last time, but am feeling pretty defeated at this point. Will post something here if I come up with a working solution.

https://github.com/supabase/supabase/issues/1318#issuecomment-1912936617

**Comment by MANTENN** [comment]
I did try to import the data via ogr2ogr:

```
brew ogr2ogr -f PostgreSQL \
  "$PGURL" \
  "$HOME/Downloads/tlgdb_2025_a_us_addr.gdb" \
  Address_Ranges \
  -nln tiger_data.addr \
  -lco GEOMETRY_NAME=geom \
  -t_srs EPSG:4269 \
  -progress

ERROR 1: ERROR:  permission denied for schema tiger_data
LINE 1: CREATE TABLE "tiger_data"."addr" ( "objectid" SERIAL, PRIMAR...
                     ^

ERROR 1: CREATE TABLE "tiger_data"."addr" ( "objectid" SERIAL, PRIMARY KEY ("objectid"), "tlid" INTEGER, "fromhn" VARCHAR(12), "tohn" VARCHAR(12), "side" VARCHAR(1), "zip" VARCHAR(5), "plus4" VARCHAR(4), "fromtyp" VARCHAR(1), "totyp" VARCHAR(1), "arid" VARCHAR(22), "mtfcc" VARCHAR(5) )
ERROR:  permission denied for schema tiger_data
LINE 1: CREATE TABLE "tiger_data"."addr" ( "objectid" SERIAL, PRIMAR...
                     ^

ERROR 1: Unable to write feature 1 from layer Address_Ranges.
ERROR 1: Terminating translation prematurely after failed
translation of layer Address_Ranges (use -skipfailures to skip errors)
```

Just a dead extension.

**Comment by mvincentong** [comment]
I am going to make a focused docs update for the PostGIS guide to clarify that postgis_tiger_geocoder requires separate TIGER data loading and that enabling the extension alone does not populate tiger_data.
